src/components/ComponentPreview.tsx

```diff
@@ -32,40 +33,42 @@ const ComponentPreviewWrapper: React.FC<ComponentPreviewWrapperProps> = ({
   const builderContext = useContext(BuilderContext);
 
   return (
-    <div className="card panel preview-panel">
-      <div className="card-header d-flex justify-content-between align-items-center">
-        <h4 className="card-title mb-0">
-          <FormattedMessage description="Component preview card title" defaultMessage="Preview" />
-        </h4>
-        <PreviewModeToggle previewMode={previewMode} setPreviewMode={setPreviewMode} />
-      </div>
+    <ErrorBoundary>
+      <div className="card panel preview-panel">
+        <div className="card-header d-flex justify-content-between align-items-center">
+          <h4 className="card-title mb-0">
+            <FormattedMessage description="Component preview card title" defaultMessage="Preview" />
+          </h4>
+          <PreviewModeToggle previewMode={previewMode} setPreviewMode={setPreviewMode} />
+        </div>
 
-      {previewMode === 'JSON' ? (
-        <JSONEditor
-          wrapperProps={{className: 'json-editor'}}
-          value={component}
-          onChange={onComponentChange}
-          theme={builderContext.theme}
-        />
-      ) : (
-        <div className="card-body">
-          <Formik
-            enableReinitialize
-            initialValues={initialValues}
-            onSubmit={() => {
-              throw new Error("Can't submit preview form");
-            }}
-          >
-            <div
-              className={clsx('component-preview', `component-preview--${component.type}`)}
-              data-testid="componentPreview"
+        {previewMode === 'JSON' ? (
+          <JSONEditor
+            wrapperProps={{className: 'json-editor'}}
+            value={component}
+            onChange={onComponentChange}
+            theme={builderContext.theme}
+          />
+        ) : (
+          <div className="card-body">
+            <Formik
+              enableReinitialize
+              initialValues={initialValues}
+              onSubmit={() => {
+                throw new Error("Can't submit preview form");
+              }}
             >
-              {children}
-            </div>
-          </Formik>
-        </div>
-      )}
-    </div>
+              <div
+                className={clsx('component-preview', `component-preview--${component.type}`)}
+                data-testid="componentPreview"
+              >
+                {children}
```

Comment:
I'd restrict the error boundary to this node:

* bugs in our own JSONEditor or Formik setup should not be suppressed, we need to be aware of them
* crashes in the preview will still allow you to fix the JSON if needed, that's currently not possible because the preview mode toggle is in the error boundary
* only the `{children}` our outside of our own control (they come from the registry)